IdentityCore/src/network/MSIDHttpRequest.m

```diff
@@ -123,7 +130,21 @@ - (void)sendWithBlock:(MSIDHttpRequestDidCompleteBlock)completionBlock
 
           if (error)
           {
-              completeBlockWrapper(nil, error);
+              if ([self.experimentBag msidBoolObjectForKey:MSID_EXP_RETRY_ON_NETWORK])
+              {
+                  [self.errorHandler handleError:error
+                                    httpResponse:nil
+                                            data:nil
+                                     httpRequest:self
+                              responseSerializer:nil
```

Comment:
Do we need to pass responseSerializer here?

Reply:
ResponseSerializer is used analyzing the httpResponse. For handling networking error, the code should not get there and there should not be any httpResponse in this case